src/examples/ui/sticky-header/css/index.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Debounce
+ *
+ * @link https://www.javascript.christmas/2020/19
+ *
+ * @param {*} func
+ * @param {*} wait
+ * @param {*} maxWait
+ * @returns
+ */
+// function debounce(func, wait, maxWait) {
+// 	let timeout;
+// 	let max;
+//
+// 	return function () {
+// 		const functionToCall = () => {
+// 			func.apply(this, arguments);
+// 			max = null;
+// 		};
+//
+// 		if (!max) {
+// 			max = setTimeout(functionToCall, maxWait);
+// 		}
+//
+// 		if (timeout) {
+// 			clearTimeout(timeout);
+// 		}
+// 		timeout = setTimeout(functionToCall, wait);
+// 	};
+// }
+
+/**
+ * Throttle
+ *
+ * @link https://www.javascript.christmas/2020/19
+ *
+ * @param {*} func
+ * @param {*} wait
+ * @returns
+ */
+function throttle(func, wait) {
+	let waiting = false;
+	return function () {
+
+		if (waiting) {
+			return;
+		}
+
+		waiting = true;
+		setTimeout(() => {
+			func.apply(this, arguments);
+			waiting = false;
+		}, wait);
+	};
+}
+
+/**
+ * Get header height
+ */
+function getOffsetHeight(element) {
+	return element ? element.offsetHeight : 0;
+}
+
+/**
+ * Set root style property
+ *
+ * @param {*} property
+ * @param {*} value
+ */
+const setRootStyleProperty = (property, value) => {
+	document.documentElement.style.setProperty(property, `${value}px`);
+};
+
+document.addEventListener('DOMContentLoaded', function () {
+
+	/**
+	 * Sticky header behavior
+	 */
+	(function () {
```

Comment:
I am used to encapsulated single behavior code like to to keep grouped and out of the global namespace. Usually this takes the form of an import and init function which would makes things more portable for other projects but none of the examples are had been set up that way.

Having said the above, I have refactored this to use an import instead of the IIFE